crates/task-impls/src/request.rs

```diff
@@ -115,12 +108,22 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>> TaskState for NetworkRequest
                 }
                 Ok(())
             }
-            HotShotEvent::QuorumProposalRequest(missing) => {
-                let ProposalMissing {
-                    view,
-                    response_chan: chan,
-                } = missing;
-                self.run_proposal(&RequestKind::Proposal(*view), chan.clone(), *view);
+            HotShotEvent::QuorumProposalRequestRecv(view_number, sender_key) => {
+                if let Some(quorum_proposal) =
+                    self.state.read().await.last_proposals().get(view_number)
+                {
+                    broadcast_event(
+                        HotShotEvent::QuorumProposalResponseSend(
+                            *view_number,
+                            sender_key.clone(),
+                            quorum_proposal.clone(),
+                        )
+                        .into(),
+                        sender,
+                    )
+                    .await;
+                }
+
```

Comment:
It's not really wrong to have this here, but it logically belongs in the "Response" task which handles responding to requests where the request task is handling sending out requests and waiting for the response